core/consensus/timeline/impl/block_appender_base.cpp

```diff
@@ -75,9 +75,9 @@ namespace kagome::consensus {
 
   outcome::result<BlockAppenderBase::SlotInfo> BlockAppenderBase::getSlotInfo(
       const primitives::BlockHeader &header) const {
-    OUTCOME_TRY(
-        slot_number,
-        babe::getSlot(header));  // TODO(xDimon): Make it consensus agnostic
+    auto consensus = consensus_selector_.get()->getProductionConsensus(header);
+    BOOST_ASSERT_MSG(consensus, "Must be returned at least fallback consensus");
+    OUTCOME_TRY(slot_number, consensus->getSlot(header));
```

Comment:
This change will fix issue #2071

Reply:
Please add Closes #2071 to this PRs description